deployment/bicep/shared/shared.bicep

```diff
@@ -0,0 +1,91 @@
+targetScope='resourceGroup'
+param location string
+param sharedResourceGroupResources object
+
+param jumpboxSubnetId string
+param agentSubnetId string
+param vmazdevopsUsername string
+param vmazdevopsPassword string
+param azureDevOpsAccount string
+
+
+param personalAccessToken string
+param resourceGroupName string
+
+
+//param environment string
+//param namePrefix string = 'not set'
+
+
+module appInsights './azmon.bicep' = {
+  name: 'azmon'
+  scope: resourceGroup(resourceGroupName)
+  params: {
+    location: location
+    sharedResourceGroupResources : sharedResourceGroupResources
+
+  }
+}
+output appInsightsConnectionString string = appInsights.outputs.appInsightsConnectionString
+
+
+module vm_devopswinvm './createvmwindows.bicep' = {
+  name: 'azdevopsvm'
+  scope: resourceGroup(resourceGroupName)
+  params: {
+    subnetId: agentSubnetId
+    username: vmazdevopsUsername
+    password: vmazdevopsPassword
+    vmName: 'azdevops-${sharedResourceGroupResources.vmSuffix}'
+    azureDevOpsAccount: azureDevOpsAccount
+    personalAccessToken: personalAccessToken
+    deployAgent: true
+  }
+}
+ 
+module vm_jumpboxwinvm './createvmwindows.bicep' = {
+  name: 'jumpboxwinvm'
+  scope: resourceGroup(resourceGroupName)
+  params: {
+    subnetId: jumpboxSubnetId
+    username: vmazdevopsUsername
+    password: vmazdevopsPassword
+    vmName: 'jumpbox-${sharedResourceGroupResources.vmSuffix}'
+  }
+}
+
+resource key_vault 'Microsoft.KeyVault/vaults@2019-09-01' = {
```

Comment:
the API version 'Microsoft.KeyVault/vaults@2021-04-01-preview' allows to set the property "enableRbacAuthorization" which would make it nicer to assign access based on either user assigned or system assigned managed identity.